factory/api/apiResolverFactory_test.go

```diff
@@ -59,8 +104,357 @@ func TestCreateApiResolver(t *testing.T) {
 		Bootstrapper:       disabled.NewDisabledBootstrapper(),
 		AllowVMQueriesChan: common.GetClosedUnbufferedChannel(),
 	}
+}
+
+func createFailingMockArgs(t *testing.T, failingSteps *failingSteps) *api.ApiResolverArgs {
+	args := createMockArgs(t)
+	coreCompStub := factory.NewCoreComponentsHolderStubFromRealComponent(args.CoreComponents)
+
+	internalMarshaller := args.CoreComponents.InternalMarshalizer()
+	coreCompStub.InternalMarshalizerCalled = func() marshal.Marshalizer {
+		failingSteps.marshallerStepCounter++
+		if failingSteps.marshallerStepCounter > failingSteps.marshallerFailingStep {
+			return nil
+		}
+		return internalMarshaller
+	}
+
+	enableEpochsHandler := args.CoreComponents.EnableEpochsHandler()
+	coreCompStub.EnableEpochsHandlerCalled = func() common.EnableEpochsHandler {
+		failingSteps.enableEpochsHandlerStepCounter++
+		if failingSteps.enableEpochsHandlerStepCounter > failingSteps.enableEpochsHandlerFailingStep {
+			return nil
+		}
+		return enableEpochsHandler
+	}
+
+	byteSliceConv := args.CoreComponents.Uint64ByteSliceConverter()
+	coreCompStub.Uint64ByteSliceConverterCalled = func() typeConverters.Uint64ByteSliceConverter {
+		failingSteps.uint64ByteSliceConvStepCounter++
+		if failingSteps.uint64ByteSliceConvStepCounter > failingSteps.uint64ByteSliceConvFailingStep {
+			return nil
+		}
+		return byteSliceConv
+	}
+
+	pubKeyConv := args.CoreComponents.AddressPubKeyConverter()
+	coreCompStub.AddressPubKeyConverterCalled = func() core.PubkeyConverter {
+		failingSteps.addressPublicKeyConverterStepCounter++
+		if failingSteps.addressPublicKeyConverterStepCounter > failingSteps.addressPublicKeyConverterFailingStep {
+			return nil
+		}
+		return pubKeyConv
+	}
+
+	args.CoreComponents = coreCompStub
+	return args
+}
+
+func TestCreateApiResolver(t *testing.T) {
+	t.Parallel()
+
+	t.Run("createScQueryService fails should error", func(t *testing.T) {
+		t.Parallel()
+
+		args := createMockArgs(t)
+		args.Configs.GeneralConfig.VirtualMachine.Querying.NumConcurrentVMs = 0
+		apiResolver, err := api.CreateApiResolver(args)
+		require.True(t, strings.Contains(err.Error(), "VirtualMachine.Querying.NumConcurrentVms"))
+		require.True(t, check.IfNil(apiResolver))
+	})
+
+	failingStepsInstance := &failingSteps{}
+	failingArgs := createFailingMockArgs(t, failingStepsInstance)
+	// do not run these tests in parallel as they all use the same args
+	t.Run("DecodeAddresses fails causing createScQueryElement error should error", func(t *testing.T) {
+		failingStepsInstance.reset()
```

Comment:
Yup, we have kind of dead code on this factories as we always check for errors when we assemble components. They should not appear if we use the real components (such as the CoreComponentsHolder). However, in order to increase the code coverage, we need to test also those error branches. Those errors can happen only on 2 situations:
1. we do something wrong in the real CoreComponentsHolder and we end up returning a nil sub-component (the code will break in a lot of other parts)
2. we do another core component. This might happen if someone else re-uses this component in another repo. 
TL;DR, I would keep these tests.